dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -0,0 +1,212 @@
+    updateList(id, index, name, 'side');
+  }
+
+  const handleServiceRemove = (index: number, name: string) => {
+    const list = { ...configList };
+    const arr = [...list[name]];
+    arr.splice(index, 1);
+    const y = { ...list, [name]: arr }
+    setConfigList(y);
+  };
+
+  const handleServiceAdd = (name: string) => {
+    let newList = { ...configList, [name]: [...configList[name], newEntry] }
+    setConfigList(newList);
+  };
+
+  const updateChart = () => {
+    dispatch(
+      renderExplorerVis({
+        tabId: tabId,
+        data: {
+          ...explorerVisualizations,
+          dataConfig: {
+            metrics: configList.metrics,
+            dimensions: configList.dimensions
+          }
+        }
+      })
+    );
+  }
+
+  const getCommonUI = (lists, sectionName: string) => lists.map((singleField, index: number) => (
+    <>
+      <div key={index} className="services">
+        <div className="first-division">
+          <EuiPanel color="subdued">
+            <EuiFormRow
+              label="Aggregation"
+              labelAppend={
+                lists.length !== 1 && (
+                  <EuiText size="xs">
+                    <EuiIcon type="cross" color="danger" onClick={() => handleServiceRemove(index, sectionName)} />
+                  </EuiText>
+                )
+              }
+            >
+              <EuiComboBox
+                aria-label="Accessible screen reader label"
+                placeholder="Select a aggregation"
+                singleSelection={{ asPlainText: true }}
+                options={AGGREGATION_OPTIONS}
+                selectedOptions={singleField.aggregation ? [{ 'label': singleField.aggregation }] : []}
+                onChange={(e) => updateList(e.length > 0 ? e[0].label : '', index, sectionName, 'aggregation')}
+
+              />
+
+            </EuiFormRow>
+            <EuiFormRow
+              label="Field"
+            >
+              <EuiComboBox
+                aria-label="Accessible screen reader label"
+                placeholder="Select a field"
+                singleSelection={{ asPlainText: true }}
+                options={fieldOptionList}
+                selectedOptions={singleField.label ? [{ 'label': singleField.label }] : []}
+                onChange={(e) => updateList(e.length > 0 ? e[0].label : '', index, sectionName, 'label')}
+              />
+            </EuiFormRow>
+
+            <EuiFormRow
+              label="Custom label"
+            >
+              <EuiFieldText
+                placeholder="Custom label"
+                value={singleField.custom_label}
+                onChange={(e) => updateList(e.target.value, index, sectionName, 'custom_label')}
+                aria-label="Use aria labels when no actual label is in use" />
+            </EuiFormRow>
+
+            {sectionName === 'metrics' && visualizations.vis.name === visChartTypes.Line || visualizations.vis.name === visChartTypes.Bar && (
```

Comment:
The condition can be saved in a separate variable.- too much logic inside the component JSX.

Reply:
created variable function to check this condition